tests/functional/n/no/no_member_assign_same_line.py

```diff
@@ -0,0 +1,21 @@
+"""https://github.com/PyCQA/pylint/issues/1555"""
+# pylint: disable=too-few-public-methods
+
+
+class ClassWithMember:
+    """Member defined in superclass."""
+    def __init__(self):
+        self.member = True
+
+
+class AssignMemberInSameLine:
+    """This class attempts to assign and access a member in the same line."""
+    def __init__(self):
+        self.member = self.member  # [no-member]
+
+
+class AssignMemberFromSuper(ClassWithMember):
+    """This assignment is valid due to inheritance."""
+    def __init__(self):
+        super().__init__()
+        self.member = self.member
```

Comment:
I would like some additional test cases just to be sure we are not emitting false postives/negatives.

What about:
```python
class AssignMemberFromSuper(ClassWithMember):
     """This assignment is valid due to inheritance."""
     def __init__(self):
         self.member = self.member
         super().__init__()
```
and
```python
class MyClass:
     """This assignment is valid due to inheritance."""
     def __init__(self):
         self.member: bool
         self.member = self.member
```

There might be more variations, but this is just what I came up with quickly.

I also wonder what happens when the assignment in the base class is conditional.
Something like:
```python
class ClassWithMember:
     """Member defined in superclass."""
     def __init__(self):
         if False:
             self.member = True

class AssignMemberFromSuper(ClassWithMember):
     """This assignment is valid due to inheritance."""
     def __init__(self):
         super().__init__()
         self.member = self.member
```
That would require control flow, but would we emit the message with the current code? Or do we assume the assignment succeeded?
If we still emit we might want to open an issue about it to track future progress.

Reply:
Good cases.

- variation in placement of call to super ✅ 
- type annotation ❌ (false negative)
- control flow with constant false values ❌ (false negative)

I've updated the PR.

Reply:
> * variation in placement of call to super ✅

👍  Sign of a well written checker 😄 

> * type annotation ❌ (false negative)

Do you think we can fix this in this PR? Seems to fit the general idea of the original issue and would prevent future issues being opened.

> * control flow with constant false values ❌ (false negative)

I think the standard is not to include tests that we do not currently pass in the test suite, but rather to open new issues for them. I've gotten this comment on some of my PRs previously. Perhaps it's best to remove these from the tests and create a new well defined issue for them. Already writing down the tests often helps with eventual implementation.

Reply:
> I think the standard is not to include tests that we do not currently pass in the test suite

Makes sense, since we don't have a way to decorate with expectedFailure given how things are set up.

> Do you think we can fix this in this PR?

I'll have a look. 🔦  Don't know much about how type annotations are tracked just yet.